src/lib/dialog/dialog-ref.ts

```diff
@@ -87,22 +90,33 @@ export class MatDialogRef<T, R = any> {
    * @param dialogResult Optional result to return to the dialog opener.
    */
   close(dialogResult?: R): void {
-    this._result = dialogResult;
+    let actualResult =  dialogResult ? dialogResult : this._defaultResult;
```

Comment:
This won't work as expected if somebody tries to close with `false` or `0`.

Reply:
Right, I'll fix this.